modules/Hoot/ui/modals/deleteFavoriteOpts.js

```diff
@@ -0,0 +1,151 @@
+                onClick: () => this.handleSubmit()
+            }
+        };
+
+        this.container = new FormFactory().generateForm( 'body', 'datasets-import-form', metadata );
+
+        this.typeInput      = this.container.select( '#optToDelete' );
+        this.submitButton   = this.container.select( '#addSubmitBtn' );
+
+        return this;
+    }
+
+    /**
+     * Validate user input to make sure it doesn't
+     * contain un-allowed characters and isn't an empty string
+     *
+     * @param d - node data
+     */
+    validateTextInput( d ) {
+        let target           = d3.select( `#${ d.id }` ),
+            node             = target.node(),
+            str              = node.value,
+
+            reservedWords    = [ 'root', 'dataset', 'dataset', 'folder' ],
+            unallowedPattern = new RegExp( /[~`#$%\^&*+=\-\[\]\\';\./!,/{}|\\":<>\?|]/g ),
+            valid            = true;
+
+        if ( reservedWords.indexOf( str.toLowerCase() ) > -1 || unallowedPattern.test( str ) ) {
+            valid = false;
+        }
+
+        if ( d.required && !str.length ) {
+            valid = false;
+        }
+
+        target.classed( 'invalid', !valid );
+        this.formValid = valid;
+        this.updateButtonState();
+    }
+
+    /**
+     * Update the form by enabling, disabling, or clearing certain
+     * fields based on the value entered
+     */
+    handleTypeChange() {
+        let selectedVal  = this.typeInput.property( 'value' ),
+            selectedType = this.getTypeName( selectedVal );
+
+        // enable input
+        if ( !selectedType ) {
+            this.submitButton.node().disabled = true;
+        } else {
+            this.submitButton.node().disabled = false;
+        }
+
+    }
+
+    /**
+     * Enable/disable button based on form validity
+     */
+    updateButtonState() {
+        let importType = this.typeInput.node().value,
+            self       = this;
+
+        this.container.selectAll( '.text-input' )
+            .each( function() {
+                let classes = d3.select( this ).attr( 'class' ).split( ' ' );
+
+                if ( classes.indexOf( 'invalid' ) > -1 || !importType.length ) {
+                    self.formValid = false;
+                }
+            } );
+
+        this.submitButton.node().disabled = !this.formValid;
+    }
+
+    /**
+     * Get the selected import-type's value
+     *
+     * @param title - title of selected import-type
+     * @returns {boolean|string} - value of type if found. otherwise, false.
+     */
+    getTypeName( title ) {
+        let allOpts = this.favorites,
+            match     = _find( allOpts, o => o.name === title );
+
+        return match.name ? match.name : false;
+    }
+
+    handleSubmit() {
+        let optName = this.typeInput.property( 'value' );
+
+        let toDelete = _find( this.favorites, o => o.name === optName );
+
+        this.processRequest = Hoot.api.deleteFavoriteOpts( toDelete )
```

Comment:
you could also potentially just pass in the name to the delete function and use that. Would make the string you send much smaller since i dont think the value associated with the name matters much?